### PR TITLE
[meshcat] Wire up geometry names for point contact visualization

### PR DESCRIPTION
**Pull request by jwnimmer-tri**
Towards #15738.

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16515)

### CODE REVIEW

**Comment by jwnimmer-tri** [comment]
___
*[multibody/meshcat/contact_visualizer.h, line 85 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16515#-MvA1qfx8_eTudqXc1CJ:-MvA1qfx8_eTudqXc1CK:bdff81u) ([raw file](https://github.com/robotlocomotion/drake/blob/5b4cadebcdd9278e4d3d2b42878e6b15f2e4ed08/multibody/meshcat/contact_visualizer.h#L85)):*
> ```objc
>   /** Adds a ContactVisualizer and connects it to the given
>   MultibodyPlant's multibody::ContactResults-valued output port and
>   geometry::QueryObject-valued output port. */
> ```

FYI @RussTedrake @joemasterjohn @ggould-tri @sammy-tri

This PR uses a novel trick.  When we pass a `plant` to a builder function, there is no need to pass a `scene_graph` pointer as well.  We can interrogate the `builder`  to find the SceneGraph that the MbP is already connected to.  This makes the user-facing API much nicer.

I have a TODO in the code to promote that logic to a reusable helper function.

**Comment by jwnimmer-tri** [comment]
+@rpoyner-tri for feature review, please.

**Review by ggould-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16515)* status: LGTM missing from assignee rpoyner-tri(platform), needs at least two assigned reviewers, missing label for release notes (waiting on @rpoyner-tri)

___
*[multibody/meshcat/contact_visualizer.h, line 85 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16515#-MvA1qfx8_eTudqXc1CJ:-MvJMael86r98z87jGME:b-rvklpg) ([raw file](https://github.com/robotlocomotion/drake/blob/5b4cadebcdd9278e4d3d2b42878e6b15f2e4ed08/multibody/meshcat/contact_visualizer.h#L85)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

FYI @RussTedrake @joemasterjohn @ggould-tri @sammy-tri

This PR uses a novel trick.  When we pass a `plant` to a builder function, there is no need to pass a `scene_graph` pointer as well.  We can interrogate the `builder`  to find the SceneGraph that the MbP is already connected to.  This makes the user-facing API much nicer.

I have a TODO in the code to promote that logic to a reusable helper function.
</blockquote></details>

TL;DR:  No defect

I've argued back and forth with myself about whether this needs to be called out in a comment, so here's my reasoning for why I _don't_ think this needs more explanation.
 * There is -- intentionally -- no intrinsic 1:1 relationship between sg and mbp.  Even if 99.99% of the time that's how it works.
 * There is aiui exactly one place where we impose an at-most-one relationship:  The input port of mbp.
 * But that's where this implementation reads from.  So is that right?  Yes, this choice means that exactly the sg providing collision results to the mbp is subject to visualization.
 * ie, the only real purpose of this backhand nonlocality is to better characterize the inputs to the mbp.
 * So in practice there can never be any ambiguous or counterintuitive behaviour here that would cause a user to seek out documentation.

Although spooky this is all perfectly correct.

**Review by rpoyner-tri — COMMENTED** [review]
:lgtm: test question

> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16515#-:-MvJw65q3ttuROIsp3T5:bipe751)* status: 1 unresolved discussion, needs at least two assigned reviewers, missing label for release notes (waiting on @jwnimmer-tri)

___
*[multibody/meshcat/contact_visualizer.cc, line 136 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16515#-MvJvtfc2E6RinTHC6Tb:-MvJvtfc2E6RinTHC6Tc:b-tfbvf2) ([raw file](https://github.com/robotlocomotion/drake/blob/5b4cadebcdd9278e4d3d2b42878e6b15f2e4ed08/multibody/meshcat/contact_visualizer.cc#L136)):*
> ```cpp
>   } else {
>     return AddToBuilder(
>         builder, plant.get_contact_results_output_port(),
> ```

This branch is not reached by unit tests. Should I worry?

**Review by jwnimmer-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16515)* status: needs at least two assigned reviewers, missing label for release notes (waiting on @jwnimmer-tri)

___
*[multibody/meshcat/contact_visualizer.cc, line 136 at r1](https://reviewable.io/reviews/robotlocomotion/drake/16515#-MvJvtfc2E6RinTHC6Tb:-MvK6CK14CAIcfgR8VdB:bam30eu) ([raw file](https://github.com/robotlocomotion/drake/blob/5b4cadebcdd9278e4d3d2b42878e6b15f2e4ed08/multibody/meshcat/contact_visualizer.cc#L136)):*
<details><summary><i>Previously, rpoyner-tri (Rick Poyner (rico)) wrote…</i></summary><blockquote>

This branch is not reached by unit tests. Should I worry?
</blockquote></details>

Done (added).

**Comment by jwnimmer-tri** [comment]
+@xuchenhan-tri for platform review per schedule (on Wednesday), please.

**Review by xuchenhan-tri — COMMENTED** [review]
:lgtm: platform

> Reviewed 4 of 5 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16515#-:-MvKRH5wBNy4lSfpw0dW:bcpz082)* status: 1 unresolved discussion (waiting on @jwnimmer-tri)

___
*[multibody/meshcat/contact_visualizer.h, line 77 at r2](https://reviewable.io/reviews/robotlocomotion/drake/16515#-MvKQ3EB6VJ4ji0D5IFg:-MvKQ3ECBQ2dd6anUfK_:bni9aky) ([raw file](https://github.com/robotlocomotion/drake/blob/ea6b59f22bd2626a14c669ca580fc150f1530491/multibody/meshcat/contact_visualizer.h#L77)):*
> ```objc
> 
>   /** Returns the geometry::QueryObject-valued input port. It should be
>   (optionally) connected to SceneGraphs's get_query_output_port(). Failure
> ```

nit, SceneGraphs's -> SceneGraph's

**Review by rpoyner-tri — COMMENTED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/16515)* status: :shipit: complete! all discussions resolved, LGTM from assignees rpoyner-tri(platform),xuchenhan-tri (waiting on @jwnimmer-tri)
